lib/sinon/match.js

```diff
@@ -109,7 +107,7 @@ var TYPE_MAP = {
 };
 
 function match(expectation, message) {
-    var m = create(matcher);
+    var m = Object.create(matcher);
```

Comment:
The whole code looks much better and less error prone by using a native function instead of reimplementing it.
Well done!